src/compiler/emitter.ts

```diff
@@ -3033,8 +3033,7 @@ module ts {
                 return false;
             }
 
-            function indentIfOnDifferentLines(parent: Node, node1: Node, node2: Node) {
-                // Use a newline for existing code if the original had one, and we're preserving formatting.
+            function indentIfOnDifferentLines(parent: Node, node1: Node, node2: Node, valueToWriteWhenNotIndenting?: string) {

```

Comment:
Use an explicit return type, document that it returns true if the indentation was indeed emitted.